test/specification/invariantSpec.ts

```diff
@@ -0,0 +1,282 @@
+import { validateSpecificationInvariant } from "../../src/specification/inverse";
+import { Specification, Match, Label, PathCondition, ExistentialCondition } from "../../src/specification/specification";
+
+describe("specification invariant validation", () => {
+    it("should accept empty specification", () => {
+        const specification: Specification = {
+            given: [],
+            matches: [],
+            projection: { type: "composite", components: [] }
+        };
+
+        expect(() => validateSpecificationInvariant(specification)).not.toThrow();
+        expect(validateSpecificationInvariant(specification)).toBe(true);
+    });
+
+    it("should accept specification with only givens", () => {
+        const specification: Specification = {
+            given: [{ name: "p1", type: "Company" }],
+            matches: [],
+            projection: { type: "composite", components: [] }
+        };
+
+        expect(() => validateSpecificationInvariant(specification)).not.toThrow();
```

Comment:
Does this pattern call the invariant function twice per test? Can this be improved?